tests/runtimes/test_generators.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pathlib
+from contextlib import nullcontext as does_not_raise
+
+import pytest
+
+import mlrun.runtimes.generators
+
+
+@pytest.mark.parametrize(
+    "strategy,param_file,expected_generator_class,expected_error",
+    [
+        (
+            "list",
+            "hyperparams.csv",
+            mlrun.runtimes.generators.ListGenerator,
+            does_not_raise(),
+        ),
+        (
+            "grid",
+            "hyperparams.json",
+            mlrun.runtimes.generators.GridGenerator,
+            does_not_raise(),
+        ),
+        (
+            "random",
+            "hyperparams.json",
+            mlrun.runtimes.generators.RandomGenerator,
+            does_not_raise(),
+        ),
+        # no strategy, default to list
+        (
+            "",
+            "hyperparams.csv",
+            mlrun.runtimes.generators.ListGenerator,
+            does_not_raise(),
+        ),
+        # no strategy, default to grid
+        (
+            "",
+            "hyperparams.json",
+            mlrun.runtimes.generators.GridGenerator,
+            does_not_raise(),
+        ),
```

Comment:
Add a case for `list` with the `JSON` file (should work, produce 2 iterations).

Reply:
done